packages/quantum-js-cli/__test__/runner.test.js

```diff
@@ -0,0 +1,228 @@
+    })
+    test("Testing evaluateInput() with input 'toText' and an empty circuit", () => {
+        runner.evaluateInput("toText", circuit);
+        expectedText = circuit.toAmazonBraket();
+        expect(console.log).toHaveBeenCalled();
+        expect(console.log).toHaveBeenCalledWith(expectedText);
+    })
+    test("Testing evaluateInput() with input 'clear' and an empty circuit", () => {
+        console.clear = jest.fn();
+        runner.evaluateInput("clear", circuit);
+        expectedText = circuit.toAmazonBraket();
+        expect(console.clear).toHaveBeenCalled();
+    })
+})
+
+//Gate operation syntax is of the regex form /(\w+\(\s*\d+\s*,\s*\[(\s*\d+\s*,{0,1}\s*)+\](,\s*\d+\.{0,1}\d+\s*)*\))/g
+//or more easily described: 
+//'gate-symbol(moment-index, [registerIndex0, registerIndex1,...], parameter0, parameter1,...)' with white space allowed liberally
+describe("Testing various forms of gate-operation call expression and see that evaluateOperation is called", ()=> {
+    //create empty circuit.
+    test("Check that evaluateOperation is called once for input 'h(1, [1])'", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("h(1, [1])", circuit);
+        expect(circuit.get(1, 1).gate.symbol).toBe('H');
+    })
+    //messing with the white space
+    test("Check that evaluateOperation is called once for input 'h( 1 , [ 1 ])'", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("h( 1 , [ 1 ])", circuit)
+        expect(circuit.get(1, 1).gate.symbol).toBe('H');
+    })
+    //while this doesn't update the circuit, it should still call evaluateOperation which detects the flaw later.
+    test("Check that no operation is added for 'h( 1 , [ 1 ], 3)' as the Hadamard operation takes no parameters", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("h( 1 , [ 1 ], 3)", circuit)
+        expect(circuit.get(1, 1)).toBeUndefined();
+    }) 
+    test("Check that evaluateOperation is called for input 'x(1, [1, 2])'", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("x(1, [1, 2])", circuit)
+        expect(circuit.get(1, 1).gate.symbol).toBe("X");
+        expect(circuit.get(1, 2).gate.symbol).toBe("X");
+    })
+    //messing with the white space
+    test("Check that evaluateOperation is called for input 'x( 1, [ 1,   2 ]   )'", () => {
+        let circuit = quantumjs.Q();
+        runner.evaluateInput("x(1, [1, 2])", circuit)
+        expect(circuit.get(1, 1).gate.symbol).toBe("X");
+        expect(circuit.get(1, 2).gate.symbol).toBe("X");
+    })
+    describe("Check that gateSymbol(...,[...]) is valid for all gate constants in the Gate module using their nameCss value", ()=> {
+        Object.entries(quantumjs.Gate.constants).forEach(function(entry) {
+            let gate = entry[1];
+            let input = gate.nameCss + (gate.is_multi_qubit ? "(1, [1, 2])" : "(1, [1])");
+            console.log(input);
```

Comment:
I think this log is for dev purposes, so can be removed

Reply:
Good call! Removed!